.github/workflows/cross-arch-build.yml

```diff
@@ -0,0 +1,128 @@
+name: Cross-Architecture Build Test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+      - release/**
+
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.event_name == 'push' && github.sha || github.ref }}
+
+jobs:
+  linux-amd64:
+    name: "Linux AMD64"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v5
+        with:
+          submodules: true
+
+      - name: Set up Go
+        uses: actions/setup-go@v6
+        with:
+          go-version: '1.24'
+
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y build-essential
+
+      - name: Verify Go installation
+        run: |
+          go version
+          echo "GOARCH: $(go env GOARCH)"
+          echo "GOOS: $(go env GOOS)"
+
+      - name: Download Go dependencies
+        run: go mod download
+
+      - name: Run make install
+        run: make install
+        env:
+          LEDGER_ENABLED: false
```

Comment:
`LEDGER_ENABLED` is always set to false. Instead of repeating it per step maybe capture as top level `env` for all jobs?